rules/ensure-external-secrets-storage-is-in-use/raw.rego

```diff
@@ -0,0 +1,33 @@
+package armo_builtins
+import data.kubernetes.api.client as client
+import data
+
+
+
+# deny workloads that doesnt support extrnal service prodvider (secretProviderClass)
+deny[msga] {
+
+    wl := input[_]
+	wl_kinds := {"Deployment","ReplicaSet","DaemonSet","StatefulSet","Job", "Pod"}
+	wl_kinds[wl.kind]
+
+	not wl.spec.volumes[0].csi.volumeAttributes.secretProviderClass
```

Comment:
notice that most workloads have a different path to spec than pods.
also, the case of cronjob is missing

Reply:
Further research shows it should be set only on Pods. Fixed by this finding.